common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/VroPackageStore.java

```diff
@@ -175,28 +177,50 @@ public final List<Package> importAllPackages(final List<Package> vroPackages, fi
 
 			packagesToImport.forEach(pkg -> {
 				String originalPkgFilePath = pkg.getFilesystemPath();
-				String backupFilePath = this.createBackupFilePath(pkg, currentDateTimeString, backupFilesDirectory);
 
 				try {
-					pkg.setFilesystemPath(backupFilePath);
-					restClient.exportPackage(pkg, dryrun, exportConfigAttributeValues, exportConfigSecureStringValues);
+					logger.info("Package for back up: ", pkg.getFQName());
+
+					List<Package> samePackagesInDest = new ArrayList<Package>();
+					for (int i = 0; i < destinationEndpointPackages.size(); i++) {
+						String currentVroPackageName = destinationEndpointPackages.get(i).getName();
+
+						if (currentVroPackageName.equals(pkg.getName())) {
+							samePackagesInDest.add(destinationEndpointPackages.get(i));
+						}
+					}
+
+					if (!samePackagesInDest.isEmpty()) {
+						List<Package> allCurrentPkgVersions = this.getPackages();
```

Comment:
Is that going to retrieve all packages available in vRO (including the system packages) or just the packages matching the one we try to import?

Reply:
hi,  just the packages matching the one we try to import

Reply:
it is going to retrieve all packages and all of their versions

Reply:
ok, we need to update this - it needs to get all package versions, but only for the packages that we are trying to import (if existing packages are present on the env).

as example if we want to import `com.vmware.pscoe.templates.buildtoolsforvmwareariasamples.xml-1.0.11-SNAPSHOT.package`, then we should be exporting any package that starts with `com.vmware.pscoe.templates.buildtoolsforvmwareariasamples.xml`

PS: Keep in mind that in the package name we CAN have dash. (so we first remove any -SNAPSHOT text, then get the last dash in the string and extract the package name based on this)